src/collectors/snmpinterfacepoll/snmpinterfacepoll.py

```diff
@@ -73,11 +73,15 @@ def publish_metrics(self, gaugesList, countersList, timestamp):
 
         for gaugeDesc in gaugesList:
             metricPath, metricValue = gaugeDesc
-            metric = Metric(
-                            metricPath, metricValue,
-                            raw_value=None, timestamp=timestamp
-                            )
-            self.publish_metric(metric)
+            # Fixes devices that return no value
+            try:
+                metric = Metric(
+                                metricPath, metricValue,
+                                raw_value=None, timestamp=timestamp
+                                )
+                self.publish_metric(metric)
+            except AttributeError:
```

Comment:
This asserts that the only possible exception that can be thrown by these operations is an AttributeError. What if the network is down and you can't publish? That might be covered in the publish_metric function - just double-checking.